R/get_coverage_wcs.R

```diff
@@ -46,35 +47,56 @@
 #' }
 #'
 get_coverage_wcs <- function(
-    wcs = c("dtm", "dsm", "omz", "omw"),
+    wcs = c("dtm", "dsm", "omz", "omw", "dhmv"),
     bbox,
     layername,
     resolution,
-    wcs_crs = "EPSG:4258",
+    wcs_crs = c("EPSG:4258", "EPSG:31370"),
     output_crs = "EPSG:31370",
     bbox_crs = "EPSG:31370",
     version = c("1.0.0", "2.0.1"),
     ...) {
+
   # prelim check
   version <- match.arg(version)
+  wcs <- tolower(wcs) # case insensitive wcs
   wcs <- match.arg(wcs)
   wcs_crs <- match.arg(wcs_crs)
   bbox_crs <- match.arg(bbox_crs)
 
+
+  # constrain version | wcs
+  if (wcs == "dhmv") {
+    assert_that(version == "1.0.0",
+      msg = "WCS `DHMV` is incompatible with versions other than `1.0.0`.")
+  }
+
   # set url
   wcs <- switch(wcs,
     omz = "https://geo.api.vlaanderen.be/oi-omz/wcs",
     omw = "https://geo.api.vlaanderen.be/oi-omw/wcs",
     dtm = "https://geo.api.vlaanderen.be/el-dtm/wcs",
-    dsm = "https://geo.api.vlaanderen.be/el-dsm/wcs"
+    dsm = "https://geo.api.vlaanderen.be/el-dsm/wcs",
+    dhmv = "https://geo.api.vlaanderen.be/DHMV/wcs"
+  )
+
+  # assure package availability
+  stopifnot(
+    assertthat = require("assertthat", quietly = TRUE),
+    httr = require("httr", quietly = TRUE),
+    sf = require("sf", quietly = TRUE),
+    terra = require("terra", quietly = TRUE)
   )
 
+  # data type assertions
   assert_that(is.character(layername))
   assert_that(is.character(output_crs))
   assert_that(inherits(bbox, "bbox"))
 
-  assert_that(is.numeric(resolution))
+  # resolution <=0 will give a `404`
+  assert_that(is.numeric(resolution) && resolution > 0)
 
+  # assemble the bounding box
```

Comment:
thanks for adding these comments - it helps with interpretation and readability of the code 👍